packages/generator/src/edmx-to-vdm/common/entity.ts

```diff
@@ -90,9 +93,20 @@ function properties(
       entity.entitySet.Name,
       p.Name
     );
-    const isComplex = isComplexType(p.Type);
     const isCollection = isCollectionType(p.Type);
-    const typeMapping = getTypeMappingEntityProperties(p.Type, complexTypes);
+    const isComplex = isCollection
+      ? isComplexType(parseCollectionTypeName(p.Type), complexTypes)
```

Comment:
Minor: It is just a personal taste but I try to avoid mutiple calls like `parseCollectionTypeName` So I woul introdoced a `const parsed = isCollection ? arseCollectionTypeName(p.Type) : pType. And then call the `isEnumType(parsed)` and `isComplexType(parsed). Then you also have one tenary operator.